cmd/dump/cmd-to-md.go

```diff
@@ -0,0 +1,247 @@
+			return fmt.Errorf("error while writing to buffer: %w", err)
+		}
+
+		if _, err := buf.WriteString(strings.ReplaceAll(cmd.Long, "\t", "  ") + "\n\n"); err != nil {
+			return fmt.Errorf("error while writing to buffer: %w", err)
+		}
+	}
+
+	if cmd.Runnable() {
+		if _, err := buf.WriteString(fmt.Sprintf("## Usage\n\n```bash\n%s\n```\n\n", strings.ReplaceAll(cmd.UseLine(), "\t", "  "))); err != nil {
+			return fmt.Errorf("error while writing to buffer: %w", err)
+		}
+	}
+
+	if len(cmd.Example) > 0 {
+		if _, err := buf.WriteString(fmt.Sprintf("## Examples\n\n```bash\n%s\n```\n\n", strings.ReplaceAll(cmd.Example, "\t", "  "))); err != nil {
+			return fmt.Errorf("error while writing to buffer: %w", err)
+		}
+	}
+
+	if err := printOptions(buf, cmd, name); err != nil {
+		return err
+	}
+
+	if hasSeeAlso(cmd) {
+		if _, err := buf.WriteString("## See Also\n\n"); err != nil {
+			return fmt.Errorf("error while writing to buffer: %w", err)
+		}
+
+		if cmd.HasParent() {
+			parent := cmd.Parent()
+			pname := parent.CommandPath()
+			link := pname + markdownExtension
+			link = strings.ReplaceAll(link, " ", "_")
+
+			if _, err := buf.WriteString(fmt.Sprintf("* [%s](%s) - %s\n", pname, linkHandler(link), parent.Short)); err != nil {
+				return fmt.Errorf("error while writing to buffer: %w", err)
+			}
+
+			cmd.VisitParents(func(c *cobra.Command) {
+				if c.DisableAutoGenTag {
+					cmd.DisableAutoGenTag = c.DisableAutoGenTag
+				}
+			})
+		}
+
+		children := cmd.Commands()
+		sort.Sort(byName(children))
+
+		for _, child := range children {
+			if !child.IsAvailableCommand() || child.IsAdditionalHelpTopicCommand() {
+				continue
+			}
+
+			cname := name + " " + child.Name()
+			link := cname + markdownExtension
+			link = strings.ReplaceAll(link, " ", "_")
+
+			if _, err := buf.WriteString(fmt.Sprintf("* [%s](%s) - %s\n", cname, linkHandler(link), child.Short)); err != nil {
+				return fmt.Errorf("error while writing to buffer: %w", err)
+			}
+		}
+	}
+
+	if _, err := buf.WriteTo(w); err != nil {
+		return fmt.Errorf("error while writing contents to file: %w", err)
+	}
+
+	return nil
+}
+
+// GenMarkdownTree will generate a markdown page for this command and all
+// descendants in the directory given. The header may be nil.
+// This function may not work correctly if your command names have `-` in them.
+// If you have `cmd` with two subcmds, `sub` and `sub-third`,
+// and `sub` has a subcommand called `third`, it is undefined which
+// help output will be in the file `cmd-sub-third.1`.
+func GenMarkdownTree(cmd *cobra.Command, dir string) error {
+	identity := func(s string) string { return s }
+	emptyStr := func(_ string) string { return "" }
+
+	return GenMarkdownTreeCustom(cmd, dir, emptyStr, identity)
+}
+
+// GenMarkdownTreeCustom is the same as GenMarkdownTree, but
+// with custom filePrepender and linkHandler.
+func GenMarkdownTreeCustom(cmd *cobra.Command, dir string, filePrepender, linkHandler func(string) string) error {
```

Comment:
I would leave this unexported and call it `genMarkdownTree`, to highlight that this is the underlying implementation for `GenMarkdownTree`.